frontend/templates/product/sections/CrossSellSection/useOptimisticAddToCart.tsx

```diff
@@ -41,7 +41,7 @@ export function useOptimisticAddToCart({
                itemcode: selectedVariant.sku,
                shopifyVariantId: selectedVariant.id,
                quantity: 1,
-               imageSrc: selectedVariant.image?.url,
+               imageSrc: selectedVariant.image?.url || product.images[0]?.url,
```

Comment:
are we guaranteed to have images have at least 1 index?

Reply:
No, but in JS `[][0]` evaluates to undefined so it would end up being an empty image like below

<img src="https://private-user-images.githubusercontent.com/66840433/238491454-9f5afcb3-d09a-402f-be0c-5d9b0129e661.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJrZXkiOiJrZXkxIiwiZXhwIjoxNjg0MzQwMjUxLCJuYmYiOjE2ODQzMzk5NTEsInBhdGgiOiIvNjY4NDA0MzMvMjM4NDkxNDU0LTlmNWFmY2IzLWQwOWEtNDAyZi1iZTBjLTVkOWIwMTI5ZTY2MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMwNTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMDUxN1QxNjEyMzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04Nzk1ZGMyMDdlZDczNDUxNjY0ZDkwOTU2MzU1ZDc2MjFhZDE0ZjE0MWRiZTg5ZDQ1MzE2MmU0MjdhNzM3YmQ2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Tobc75LrZJwk4IPPIqz9GzLJ9JoNyEk6eQvgoD3rqMY">